src/functional/ffront/func_to_foast.py

```diff
@@ -93,9 +96,15 @@ def _preprocess_definition_ast(cls, definition_ast: ast.AST) -> ast.AST:
 
     @classmethod
     def _postprocess_dialect_ast(
-        cls, foast_node: foast.FieldOperator, annotations: dict[str, Any]
+        cls,
+        foast_node: foast.FieldOperator,
```

Comment:
```suggestion
        foast_node: foast.FieldOperator | foast.ScanOperator,
```
If it creates problems with mypy down the line skip it as unrelated to this PR.

Reply:
This needs to be changed for the regular type deduction pass as well. I'd leave this for another fix PR.